src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/CastHelpers.cs

```diff
@@ -316,6 +319,10 @@ internal static unsafe class CastHelpers
             return ChkCastClassSpecial(toTypeHnd, obj);
         }
 
+        [DebuggerHidden]
+        private static bool IsAssignable(void* fromTypeHnd, void* toTypeHnd)
```

Comment:
- Should we use consistent name? Ether `IsAssignable` or `AreTypeAssignable` is fine me - as long it is used consistently.

- Should this follow the same pattern as other casting helper and do a lookup in cast-cache first before calling the slow "NoCache" path?

Reply:
> Should we use consistent name?

We are consistent. The JIT helper is called `IsAssignable()` for both versions (CoreCLR and native AOT), and both call an `AreTypesAssignable()` method as their implementation.

> Should this follow the same pattern as other casting helper and do a lookup in cast-cache first before calling the slow "NoCache" path?

Currently, both native AOT and CoreCLR cache, but the caching is done within the underlying `AreTypesAssignable` implementations. For native AOT the cache can be observed in the same file in `AreTypesAssignableInternal`. For CoreCLR, the caching is done within the implementation of `TypeHandle::CanCastTo()`

Reply:
> The JIT helper is called IsAssignable() for both versions (CoreCLR and native AOT), and both call an AreTypesAssignable() method as their implementation.

I would call the JIT helper `AreTypesAssignable` as well.

> For CoreCLR, the caching is done within the implementation of TypeHandle::CanCastTo()

This implementation is leaving performance on the table. All existing managed casting helpers in this file are doing the cache lookup on the managed side to avoid paying managed/unmanaged transition costs for cache hit, and only call the NoCacheLookup path with managed/unmanaged transition path on cache miss.